ctapipe/plotting/charge_resolution.py

```diff
@@ -0,0 +1,358 @@
+import numpy as np
+import pandas as pd
+from matplotlib.ticker import FuncFormatter
+from matplotlib import pyplot as plt
+import os
+from ctapipe.core import Component
+from traitlets import Unicode, Int
+
+
+def sum_errors(array):
+    """
+    Simple sum of squares to combine errors
+
+    Parameters
+    ----------
+    array : ndarray
+
+    Returns
+    -------
+    float
+    """
+    return np.sqrt(np.sum(np.power(array, 2)) / array.size)
```

Comment:
I was wondering why here `np.power(array, 2)` was used instead of `array**2`. I hardly ever saw the former version and find the latter a little bit simpler to read. So I assumed the former version must be somehow superior, I assumed it was maybe faster ... so I looked at the execution time ... 
```python
import numpy as np
a = np.random.normal(size=100000)
assert (np.power(a, 2) == a**2).all()  # assert there is no difference in the result
%timeit a**2  # 55 µs ± 318 ns
%timeit np.power(a, 2)  # 692 µs ± 6.09

b = (np.random.normal(size=100000) * 200).astype('i4')
assert (np.power(b, 2) == b**2).all()
%timeit b**2   # 25.7 µs ± 285 ns
%timeit np.power(b, 2)  # 182 µs ± 1.3 µs
```

So it seems on my machine `np.power()` is a bit slower than `**`, but returns the same result. So I am a bit puzzled what the reason for using `np.power` here might be? (maybe it performs different on different python and numpy versions?) 

Please note, I'm not trying to say one should use one version or the other due to performance differences. I think performance should be optimized last. So if `np.power` is used here for improved readability or for other reasons I believe it is totally fine. I am just curious.

Reply:
Hi, thats weird I alway kind of assumed internally these are calling the same function and would take the same time! I'd be kind interested to find out what is happening internally there to cause the difference. 
Personally I use np.power because I am used to seeing C code and it is a similar syntax so helps readability for me, but I have no strong feelings either way.

Reply:
Ah .. so the `pow()` in C is also slower than simple multiplication, see for example https://baptiste-wicht.com/posts/2017/09/cpp11-performance-tip-when-to-use-std-pow.html

The same principle applies of course to CPython, so *pow()*, which is named `np.power()`, should be slower than simple multiplication, but it is able to work with arbitrary powers. 

It seems the `**` operator, while being able to work with arbitrary powers just like `pow`, tries to be smart about the special case `power==2`, where it does not delegate to the lower level `np.power()` function but to simple multiplication. However already at `power==3` both have exactly the same behaviour, see below:
```python
In [1]: import numpy as np

In [2]: a = np.random.normal(size=1000) + 5  # only positive values in a

In [3]: %timeit a*a
872 ns ± 3.8 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [4]: %timeit a*a*a
1.81 µs ± 11.4 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [5]: # simple multiplication is optimized in contrast to arbitrary powers, see below

In [6]: %timeit np.power(a, 2)
8.29 µs ± 21.1 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)

In [7]: %timeit np.power(a, 3)
56.4 µs ± 172 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)

In [8]: # but simple multiplication cannot do this:

In [9]: %timeit np.power(a, 2.5)
56.3 µs ± 129 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)

In [10]: # Pythons ** operator seems to delegate to simple multiplication in case the exp == 2

In [11]: %timeit a**2
1.56 µs ± 7.79 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [12]: %timeit a**3
56.7 µs ± 218 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)

In [13]: # the statement above is exactly as slow as np.power(a, 3)
``` 

So as a rule of thumb (for Python), I would say: 

> Always use `base**exp` instead of `np.power(base, exp)`. 

(I personally would despite the numbers above not start optimizing runtime using multiplication for higher exponents, unless it is proven that this is really the bottleneck.)